### DISCUSSION TITLE
Best way for custom requests via pb js?

### DISCUSSION THREAD

**Discussion by recoilme** [pocketbase/pocketbase#686] [Q&A] [answered]
Hello. Nice project!

I use pb as library. I created some custom routes and want use provided js library for requests. Please suggest best way how to do that?

**Comment by ganigeorgiev** [comment, marked as answer]
You can attach your own helper request functions to the `PocketBase.prototype` chain and call `this.send(path, requestOptions)`. something like:

```js
PocketBase.prototype.myCustomAction = function(bodyData = {}, queryParams = {}) {
    return this.send('/path/to/my/custom/action', {
        'method': 'POST',
        'params': queryParams,
        'body':   bodyParams,
        'heades': {
            // some extra headers...
        }
    });
}

const client = new PocketBase(...);

await client.myCustomAction();
```

You can see how `send` works in https://github.com/pocketbase/js-sdk/blob/master/src/Client.ts#L214.

_I'll move the issue to Q&A discussion._

**Reply by recoilme** [reply]
May be add some base methods for custom requests?
```
client.customGetRequest(path)
client.customPostRequest(path,body)
```

**Reply by ganigeorgiev** [reply]
@recoilme But that's the same as `client.send()`.

The above was just an example how to attach your own named helpers, but you can always just call:
```js
await client.send(path, requestConfig);
```
_(where `requestConfig` is the fetch() request options per https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#supplying_request_options)_

**Reply by recoilme** [reply]
Got it! Thank you for reply, i will try
